### PR TITLE
feat: ✨  find files in jsonnet

### PR DESCRIPTION
**Pull request by ghost**
this stage, we find our target object types and print them to the stderr. Before implementing writing to disk, we need some safety mechanisms so bad things don't happen, such as deleting whole directories and allowing access to parent directories, etc

### CODE REVIEW

**Comment by ghost** [comment]
test run output:

```
└─> JSONNET_PATH=$(pwd)/vendor go run ../../cmd/jsonnet-filer;
jsonnet-filer version: dev none unknown unknown
[]v1alpha1.File{
	{
		ApiVersion:       "jsonnet-filer.zeet.co/v1alpha1",
		Kind:             "File",
		Metadata:         v1alpha1.ObjectMeta{Name: "gen/kube/deployment.yml"},
		Content:          map[string]interface{}{},
		EncodingStrategy: "yaml",
	},
	{
		ApiVersion: "jsonnet-filer.zeet.co/v1alpha1",
		Kind:       "File",
		Metadata:   v1alpha1.ObjectMeta{Name: "gen/local/k3d.yml"},
		Content: map[string]interface{}{
			"apiVersion": "k3d.io/v1alpha4",
			"image":      "rancher/k3s:v1.24.7-k3s1-arm64",
			"kind":       "Simple",
			"metadata":   map[string]interface{}{"name": "personal"},
		},
		EncodingStrategy: "yaml",
	},
	{
		ApiVersion: "jsonnet-filer.zeet.co/v1alpha1",
		Kind:       "File",
		Metadata:   v1alpha1.ObjectMeta{Name: "Taskfile.yml"},
		Content: map[string]interface{}{
			"tasks": map[string]interface{}{
				"k3d:down": map[string]interface{}{"cmds": []interface{}{"k3d cluster delete --config gen/local/k3d.yml"}},
				"k3d:up": map[string]interface{}{
					"cmds":   []interface{}{"k3d cluster create --config gen/local/k3d.yml"},
					"status": []interface{}{"k3d cluster get personal"},
				},
			},
			"version": "3",
		},
		EncodingStrategy: "yaml",
	},
}
writing file: /Users/wes/projects/jsonnet-filer/tests/various/gen/kube/deployment.yml
apiVersion: jsonnet-filer.zeet.co/v1alpha1
content: {}
encodingStrategy: yaml
kind: File
metadata:
  name: gen/kube/deployment.yml

-----------
writing file: /Users/wes/projects/jsonnet-filer/tests/various/gen/local/k3d.yml
apiVersion: jsonnet-filer.zeet.co/v1alpha1
content:
  apiVersion: k3d.io/v1alpha4
  image: rancher/k3s:v1.24.7-k3s1-arm64
  kind: Simple
  metadata:
    name: personal
encodingStrategy: yaml
kind: File
metadata:
  name: gen/local/k3d.yml

-----------
writing file: /Users/wes/projects/jsonnet-filer/tests/various/Taskfile.yml
apiVersion: jsonnet-filer.zeet.co/v1alpha1
content:
  tasks:
    k3d:down:
      cmds:
      - k3d cluster delete --config gen/local/k3d.yml
    k3d:up:
      cmds:
      - k3d cluster create --config gen/local/k3d.yml
      status:
      - k3d cluster get personal
  version: "3"
encodingStrategy: yaml
kind: File
metadata:
  name: Taskfile.yml

-----------
```

File: pkg/api/v1alpha1/file.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package v1alpha1
+
+const (
+	ApiVersion string = "jsonnet-filer.zeet.co/v1alpha1"
+	Kind       string = "File"
+)
+
+type ObjectMeta struct {
```
Comment (zzh8829) [inline pkg/api/v1alpha1/file.go:8]: would it be worth while to implement these using upstream types like 
```
import (
    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)


type File struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Content          any        `json:"content"`
	EncodingStrategy string     `json:"encodingStrategy"`
}
```

Comment (ghost) [reply pkg/api/v1alpha1/file.go:8]: Possible in a future iteration. Right now, I don't want to add fields that have no meaning, such as annotations, etc, nor do I want this to be strictly tied to k8s. (similarities help people though).

**Review by zzh8829 — APPROVED** [review]
looks good

**Pull request merged**